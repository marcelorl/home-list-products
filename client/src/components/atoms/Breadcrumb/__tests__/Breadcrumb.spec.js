import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../Breadcrumb.fixture'

const { mount, getWrapper, get } = createTestContext({fixture})

beforeEach(mount)

describe('#Breadcrumb atom', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('should have 3 items', () => {
    expect(getWrapper('ul li')).toHaveLength(3)
  })

  it('should have labels and links passed through props', () => {
    const items = get('props').items
    const li = getWrapper('ul li')

    expect(li.at(0).text()).toEqual(items[0].label)
    expect(li.at(0).find('a').prop('href')).toEqual(items[0].link)

    expect(li.at(1).text()).toEqual(items[1].label)
    expect(li.at(1).find('a').prop('href')).toEqual(items[1].link)

    expect(li.at(2).text()).toEqual(items[2].label)
    expect(li.at(2).find('a').prop('href')).toEqual(items[2].link)
  })
})
