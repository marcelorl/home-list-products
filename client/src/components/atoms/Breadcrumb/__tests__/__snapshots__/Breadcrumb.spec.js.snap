// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Breadcrumb atom renders component 1`] = `
<Breadcrumb
  items={
    Array [
      Object {
        "label": "Início",
        "link": "/inicio",
      },
      Object {
        "label": "Homem",
        "link": "/homem",
      },
      Object {
        "label": "Novidades Internacionais",
        "link": "/",
      },
    ]
  }
>
  <styled.ul
    className="breadcrumb"
  >
    <ul
      className="breadcrumb sc-bdVaJa Bzosa"
    >
      <styled.li
        key="Início"
      >
        <li
          className="sc-bwzfXH ihlxUm"
        >
          <a
            href="/inicio"
          >
            Início
          </a>
        </li>
      </styled.li>
      <styled.li
        key="Homem"
      >
        <li
          className="sc-bwzfXH ihlxUm"
        >
          <a
            href="/homem"
          >
            Homem
          </a>
        </li>
      </styled.li>
      <styled.li
        key="Novidades Internacionais"
      >
        <li
          className="sc-bwzfXH ihlxUm"
        >
          <a
            href="/"
          >
            Novidades Internacionais
          </a>
        </li>
      </styled.li>
    </ul>
  </styled.ul>
</Breadcrumb>
`;
